ignite/trainer/trainer.py

```diff
@@ -133,9 +135,9 @@ def _fire_event(self, event_name):
                 func(self, *args, **kwargs)
 
     def _train_one_epoch(self, training_data):
+        self.iterations_per_epoch = len(training_data)
```

Comment:
I'm unsure about the best way to proceed in here, if it's best to store the whole `training_data` in `self` or only it's length. @alykhantejani what are your thoughts?

Reply:
`training_data` was stored with the previous API [ref](https://github.com/pytorch/ignite/blob/707d0ecb31db4283f93dc49135b05f30b2515356/ignite/trainer/trainer.py#L68)

Reply:
I think it might be better to store the ref to the whole `training_data` or `validation_data` and then in logging we can just call `len` on this. 

Once #54 is merged (after this), this will likely change again so not too fussy right now. I think once #54 is in the `Engine` will have a `dataset` field or something similar.